### PR TITLE
add new freamwork: cinatra

### PR DESCRIPTION
**Pull request by qicosmos**

### CODE REVIEW

**Comment by nbrady-techempower** [comment]
Hi @qicosmos - Welcome and thanks for your first contribution!

The test implementation code needs to live in this repo for review. Please update to include that here instead of pulling it from a different source.

**Comment by qicosmos** [comment]
@nbrady-techempower 
Thanks for your reply, i have updated the pr, please check it again.

File: frameworks/C++/cinatra/cinatra.dockerfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+RUN git clone https://github.com/chriskohlhoff/asio.git
+
+WORKDIR $IASIO
+
+RUN ./autogen.sh && ./configure
+RUN make -j && make install
+
+WORKDIR /
+
+RUN git clone https://github.com/qicosmos/cinatra.git
```
Comment (nbrady-techempower) [inline frameworks/C++/cinatra/cinatra.dockerfile]: Please `git checkout` a specific release or git hash for all repos.

File: frameworks/C++/cinatra/cinatra.dockerfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
+FROM ubuntu:18.04
+
+RUN apt-get update -yqq && \
+	apt-get install -yqq cmake git uuid-dev gcc g++ autoconf
+
+ENV IASIO=/asio/asio
+ENV CINATRA_HOME=/cinatra
+ENV CINATRA=/cinatra/example
+
+RUN git clone https://github.com/chriskohlhoff/asio.git
```
Comment (nbrady-techempower) [inline frameworks/C++/cinatra/cinatra.dockerfile]: This one as well

File: frameworks/C++/cinatra/cinatra.dockerfile (hunk 1 above)
Comment (qicosmos) [reply frameworks/C++/cinatra/cinatra.dockerfile]: ok, have done.

**Pull request merged**